### ISSUE TITLE
[CT-2873] [Feature] `dbt init` should create project in current dir (`.`) instead of creating a new dir

### ISSUE THREAD

**Issue by dataders** [dbt-labs/dbt-core#8216]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

TL;DR: I want `dbt init` to behave like `git init` and by default work within the current directory rather than make a new directory.

from the `git init` ([docs page](https://git-scm.com/docs/git-init)):
> If the `$GIT_DIR` environment variable is set then it specifies a path to use instead of `./.git` for the base of the repository.

### Describe alternatives you've considered

- a command-line flag to override this directory-creation step?
- guidance that users should follow explicitly these steps in order:
  - `dbt init`
  - `cd my_new_dbt_proj_dir`
  - `git init`

### Who will this benefit?

those who need to quickly make standalone project examples and sync them to GitHub

### Are you interested in contributing this feature?

yes

### Anything else?

I noticed that the [`dbt init` docs](https://docs.getdbt.com/reference/commands/init) are missing information that's provided in the CLI via `dbt init -h`

**Comment by dbeatty10** [comment]
Agreed -- I always expected `dbt init` to use the current working directory as you described with `git init`.

But since this is such long-standing behavior, I think we'd need to use one of the alternatives you mentioned instead:

> - a command-line flag to override this directory-creation step?
> - guidance that users should follow explicitly these steps in order:
>   - `dbt init`
>   - `cd my_new_dbt_proj_dir`
>   - `git init`

Maybe a [global config](https://docs.getdbt.com/reference/global-configs/about-global-configs) option named `INIT_DIR` (so that it can get included in the global `profiles.yml`) along with `DBT_INIT_DIR` environment variable and `--init-dir` CLI flag?

> I noticed that the [`dbt init` docs](https://docs.getdbt.com/reference/commands/init) are missing information that's provided in the CLI via `dbt init -h`

Would you mind opening an issue (or a PR) in [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com)?
